netmiko/fortinet/fortinet_ssh.py

```diff
@@ -1,104 +1,298 @@
-            if self.vdoms:
-                disable_paging_commands.append("end")
-            outputlist = [
-                self._send_command_timing_str(command, last_read=3)
-                for command in disable_paging_commands
-            ]
-            # Should test output is valid
-            new_output = self.RETURN.join(outputlist)
-
-        return output + new_output
-
-    def _retrieve_output_mode(self) -> None:
-        """Save the state of the output mode so it can be reset at the end of the session."""
-        reg_mode = re.compile(r"output\s+:\s+(?P<mode>.*)\s+\n")
-        output = self._send_command_str("get system console")
-        result_mode_re = reg_mode.search(output)
-        if result_mode_re:
-            result_mode = result_mode_re.group("mode").strip()
-            if result_mode in ["more", "standard"]:
-                self._output_mode = result_mode
-
-    def cleanup(self, command: str = "exit") -> None:
-        """Re-enable paging globally."""
-        if self.allow_disable_global:
-            # Return paging state
-            output_mode_cmd = f"set output {self._output_mode}"
-            enable_paging_commands = ["config system console", output_mode_cmd, "end"]
-            if self.vdoms:
-                enable_paging_commands.insert(0, "config global")
-            # Should test output is valid
-            for command in enable_paging_commands:
-                self.send_command_timing(command)
-        return super().cleanup(command=command)
-
-    def save_config(
-        self, cmd: str = "", confirm: bool = False, confirm_response: str = ""
-    ) -> str:
-        """Not Implemented"""
-        raise NotImplementedError
+from netmiko.base_connection import BaseConnection
+from netmiko import log
+from netmiko.exceptions import (
+	NetmikoTimeoutException,
+	NetmikoAuthenticationException,
+	ConfigInvalidException,
+)
+
+
+class FortinetSSH(BaseConnection):
+	def _modify_connection_params(self) -> None:
+		"""Modify connection parameters prior to SSH connection."""
+		paramiko_transport = getattr(paramiko, "Transport")
+		paramiko_transport._preferred_kex = (
+			"diffie-hellman-group14-sha1",
+			"diffie-hellman-group-exchange-sha1",
+			"diffie-hellman-group-exchange-sha256",
+			"diffie-hellman-group1-sha1",
+		)
+
+	def session_preparation(self) -> None:
+		"""Prepare the session after the connection has been established."""
+
+		data = self._test_channel_read(pattern="to accept|[#$]")
+		# If "set post-login-banner enable" is set it will require you to press 'a'
+		# to accept the banner before you login. This will accept if it occurs
+		if "to accept" in data:
+			self.write_channel("a\r")
+			self._test_channel_read(pattern=r"[#$]")
+
+		self.set_base_prompt(alt_prompt_terminator="$")
+		self.disable_paging()
+		self.clear_buffer()
+
+	def set_base_prompt(self, 
+		pri_prompt_terminator="#", 
+		alt_prompt_terminator="$", 
+		delay_factor=1.0,
+		pattern = r"(#|\$)"
```

Comment:
I'd be fine with that, why do you think it would cause an issue?

Reply:
Done

Reply:
The issue is that I later re-wrap `pattern` in another set of parenthesis (in another logical-or) and doing that causes problems in the regex behavior. One workaround to this is to use the square-bracket so that there is only one set of parenthesis.